analyzer/codechecker_analyzer/cmd/analyze.py

```diff
@@ -559,6 +559,28 @@ def __update_skip_file(args):
         shutil.copyfile(args.skipfile, skip_file_to_send)
 
 
+def __cleanup_metadata(metadata_prev, metadata):
+    """ Cleanup metadata.
+
+    If the source file for a plist does not exist, remove the plist from the
+    system and from the metadata.
+    """
+    if not metadata_prev:
+        return
+
+    result_src_files = metadata_prev['result_source_files']
+    for plist_file, source_file in result_src_files.items():
+        if not os.path.exists(source_file):
+            try:
+                LOG.info("Remove plist file '%s' because it refers to a "
+                         "source file ('%s') which was removed.",
+                         plist_file, source_file)
+                del metadata['result_source_files'][plist_file]
+                os.remove(plist_file)
```

Comment:
Consider the unlikely case of there being an exception while deleting the plist file from the filesystem. This order would result in the plist file still there, but the metadata deleted. Is this better than then the other way around (try to delete from the fs first, then if succeeded remove the metadata)? Or maybe it just doesn't matter?

Reply:
I we keep the plist files with references to missing source files, the store command will fail because the relevant files can not be collected and uploaded. If we store based on the metadata file the dangling plist file should not cause a problem just more disk space is used.
@csordasmarton do we store and collect the reports based on the metadata list or just parsing all the plist files without using the list in the metadata file?

Reply:
@gyorb We use the 'result_source_files' values at the CodeChecker parse command:

https://github.com/Ericsson/codechecker/blob/87d3e7bd13747cdf9b21426e3ca2bea99ae5abb8/analyzer/codechecker_analyzer/cmd/parse.py#L470-L476

As I remember we use this information to print that we analyzed a source file but we didn't find any bug in that file:
https://github.com/Ericsson/codechecker/blob/87d3e7bd13747cdf9b21426e3ca2bea99ae5abb8/analyzer/codechecker_analyzer/cmd/parse.py#L273